packages/lisk-cryptography/test/nacl.js

```diff
@@ -0,0 +1,238 @@
+			return expect(signedKeys.publicKeyBytes.length).to.be.eql(
+				CRYPTO_SIGN_PUBLICKEYBYTES,
+			);
+		});
+
+		it('should return privateKeyBytes with buffer of length 64', () => {
+			return expect(signedKeys.privateKeyBytes.length).to.be.eql(
+				CRYPTO_SIGN_PRIVATEKEYBYTES,
+			);
+		});
+	});
+
+	describe('#signDetached', () => {
+		let msgBytes;
+		let signatureBytes;
+		beforeEach(() => {
+			msgBytes = digestMessageStub();
+			signatureBytes = signDetached(
+				msgBytes,
+				Buffer.from(defaultPrivateKey, 'hex'),
+			);
+			return Promise.resolve();
+		});
+
+		it('should create a buffer signature', () => {
+			return expect(Buffer.from(signatureBytes).toString('hex')).to.be.eql(
+				defaultSignature,
+			);
+		});
+
+		it('should return signatureBytes with buffer of length 64', () => {
+			return expect(signatureBytes.length).to.be.eql(CRYPTO_SIGN_BYTES);
+		});
+	});
+
+	describe('#detachedVerify', () => {
+		let msgBytes;
+		beforeEach(() => {
+			msgBytes = digestMessageStub();
+			return Promise.resolve();
+		});
+
+		it('should return false if the signature is invalid', () => {
+			const verification = detachedVerify(
+				msgBytes,
+				Buffer.from(makeInvalid(defaultSignature), 'hex'),
+				Buffer.from(defaultPublicKey, 'hex'),
+			);
+			return expect(verification).to.be.false;
+		});
+
+		it('should return true if the signature is valid', () => {
+			const verification = detachedVerify(
+				msgBytes,
+				Buffer.from(defaultSignature, 'hex'),
+				Buffer.from(defaultPublicKey, 'hex'),
+			);
+			return expect(verification).to.be.true;
+		});
+	});
+
+	describe('encrypt and decrypt message with converted key pair and nonce', () => {
+		let convertedPublicKey;
+		let convertedPrivateKey;
+
+		beforeEach(() => {
+			convertedPublicKey = convertPublicKeyEd2CurveStub();
+			convertedPrivateKey = convertPrivateKeyEd2CurveStub();
+			return Promise.resolve();
+		});
+
+		describe('#box', () => {
+			let encryptedMessageBytes;
+
+			beforeEach(() => {
+				encryptedMessageBytes = box(
+					Buffer.from(defaultMessage, 'utf8'),
+					Buffer.from(defaultNonce, 'hex'),
+					convertedPublicKey,
+					convertedPrivateKey,
+				);
+				return Promise.resolve();
+			});
+
+			it('should encrypt a message', () => {
+				return expect(
+					Buffer.from(encryptedMessageBytes).toString('hex'),
+				).to.be.eql(defaultEncryptedMessage);
+			});
+		});
+
+		describe('#boxOpen', () => {
+			let decryptedMessageBytes;
+
+			beforeEach(() => {
+				decryptedMessageBytes = boxOpen(
+					Buffer.from(defaultEncryptedMessage, 'hex'),
+					Buffer.from(defaultNonce, 'hex'),
+					Buffer.from(convertedPublicKey, 'hex'),
+					Buffer.from(convertedPrivateKey, 'hex'),
+				);
+				return Promise.resolve();
+			});
+
+			it('should decrypt a message', () => {
+				return expect(
+					Buffer.from(decryptedMessageBytes).toString('utf8'),
+				).to.be.eql(defaultMessage);
+			});
+		});
```

Comment:
It would be nice to have an integration test to make sure the two play together nicely. Also potentially for sign/verify.